pkg/pkgmgr/rpm.go

```diff
@@ -361,6 +367,52 @@ func (rm *rpmManager) probeRPMStatus(ctx context.Context, toolImage string) erro
 	return nil
 }
 
+func (rm *rpmManager) generateToolInstallCmd(ctx context.Context, toolsListed *llb.State) (string, error) {
+	applicationsList, err := buildkit.ExtractFileFromState(ctx, rm.config.Client, toolsListed, "/applications.txt")
+	if err != nil {
+		return "", err
+	}
+
+	// Convert applicationsList so we have a string to work with
+	applicationsListSplit := strings.Split(string(applicationsList), "\n")
+
+	// packageManagersInstalled is the package manager(s) available within the tooling image
+	// RPM must be excluded from this list as it cannot connect to RPM repos
+	var packageManagersInstalled []string
+	packageManagerList := []string{"tdnf", "dnf", "microdnf", "yum"}
+
+	for i := range applicationsListSplit {
+		for _, packageManager := range packageManagerList {
+			if applicationsListSplit[i] == packageManager {
+				packageManagersInstalled = append(packageManagersInstalled, applicationsListSplit[i])
+			}
```

Comment:
this could potentially be a very large for loop if we are iterating each application and then each package manager for each application. Instead of iterating each application, is there a way we can simplify the applicationsListSplit string with regexp to see what matches tdnf/dnf/microdnf/yum ?

Reply:
if that doesn't work we could also create a map by iterating applicationsListSplit once and map those applications as "true" value and check for those when iterating packageManagerList